internal/stores/types.go

```diff
@@ -0,0 +1,106 @@
+package stores
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"go.sia.tech/core/types"
+	rhpv2 "go.sia.tech/renterd/rhp/v2"
+)
+
+var zeroCurrency = currency(types.ZeroCurrency)
+
+type (
+	currency       types.Currency
+	fileContractID types.FileContractID
+	publicKey      types.PublicKey
+	hostSettings   rhpv2.HostSettings
```

Comment:
All of the types in `types` (I think) have `EncodeTo` and `DecodeFrom` methods, so you could do something like:
```go
func encode(v types.EncoderTo) []byte {
    var buf bytes.Buffer
    e := types.NewEcoder(&buf)
    v.EncodeTo(e)
    e.Flush()
    return buf.Bytes()
}

func decode(b []byte, v types.DecoderFrom) error {
    d := types.NewBufDecoder(b)
    v.DecodeFrom(d)
    return d.Err()
}
```
and then e.g.
```go
func (fcid *fileContractID) Scan(value interface{}) error {
    return decode(value.([]byte), (*types.FileContractID)(fcid))
}
```

not a big deal, but it would clean things up a bit and increase the efficiency of `currency` at least